### PR TITLE
VACMS-3710: Add eslint action and update existing js to pass linting

### PR DESCRIPTION
**Pull request by acrollet**
## Description

See #3710 

## Testing done

Manually tested affected functionality.

## QA steps

As a developer, confirm that:
- [ ] ACs are met
- [x] Topics/Audiences JS works correctly (edit a checklist node, 'TAGS' section)
- [x] Contact Paragraphs JS works correctly (edit a VAMC Facility Health Service, add/edit a phone number, confirm that extension field is hidden when selecting Fax/SMS/TTY)
- [x] Service location js functions correctly (Edit a VAMC Facility Health Service, expand service locations, confirm that address field is shown when unchecking 'Use the facility's street address?')
- [x] GTM code functions correctly
- [x] Support email link functions correctly (click 'Help' in the admin toolbar)
- [x] CLP spotlight panel functions correctly (no more than 3 link teasers are allowed)
- [x] Single Alert JS functions correctly (Edit a step-by-step node, ensure that only a single reusable OR non-reusable alert may be added)
- [x] Confirm that smart date interactions JS functions correctly (Edit an event, ensure that the 'All day' checkbox is not shown in the Date and Time fieldset)
- [x] Sections Accordion js functions correctly (confirm on user profile page)
- [x] Anchor Links js functions correctly (Go to /help/product-guides/vamc/by-content-type/detail-pages and confirm that the section title is not hidden when clicking on an anchor link)
- [x] Confirm that proofing tooltip js functions correctly (View a Benefits Detail page, confirm that tooltips function correctly)
- [x] Confirm that vagovadmin theme build functions correctly (Remove all files in assets/css, run gulp and npm build and confirm that no changes are introduced) 

### Definition of Done

- [x] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [x] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `Core Application Team`
- [x] `Product Support Team`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [x] No announcement is needed for this code change. 
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/js/script.js [hunk 1]
```diff
@@ -2,26 +2,11 @@
...
-            .children(".field--type-list-string")
-            .find("select")
-            .val() === "expanding"
-            ? "block"
-            : "none";
-        });
-      });
-    },
-  };
+(($, Drupal) => {
```
Comment (acrollet) [inline docroot/modules/custom/va_gov_backend/js/script.js:5]: @kevwalsh looking at git history it's my understanding that `field_alert_trigger_text` was removed in the past and this js should no longer be necessary - can you confirm? Thanks!!

File: docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
  * @file
```
Comment (acrollet) [inline docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:2]: @ndouglas do these changes look reasonable to you? Just want to make sure that any ignorance on my part about how the GA js won't come back to bite us 😅

Comment (ndouglas) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:2]: They look reasonable to me.  The BeHat tests should pick up if anything got broken, I think.  I'll take a look-see right now.

Comment (ndouglas) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:2]: Yeah, after checking out the PR and clearing caches, the dataLayer still seems to be showing appropriate values:

Screen Shot 2020-12-11 at 1 17 45 PM

Comment (ndouglas) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:2]: reminds me, I'ma grab that `contentOwner` issue

Comment (acrollet) [reply docroot/modules/custom/va_gov_backend/js/gtm_tag_trackers.js:2]: thanks so much!

File: docroot/modules/custom/va_gov_backend/js/script.js [hunk 3]
```diff
@@ -2,86 +2,111 @@
...
...A%5BThis%20is%20a%20template.%20%20You%20can%20delete%20the%20text%20you%20don%E2%80%99t%20need%2C%20and%20feel%20free%20to%20add%20your%20own.%5D%0A%0AI%20need%20help%20with%3A%20%0A-%20A%20bug%20%2F%20defect%20I%20encountered%0A-%20Something%20that%E2%80%99s%20not%20working%20as%20I%20expected%0A-%20Learning%20how%20to%20do%20something%20in%20the%20CMS%0A%0AHere%E2%80%99s%20some%20more%20detail%20about%20what%20I%20was%20trying%20to%20do%3A%0A%5BTell%20us%20more.%20It%20helps%20to%20add%20a%20step-by-step%20list%20of%20what%20we%E2%80%99d%20need%20to%20do%20to%20reproduce%20your%20problem.%5D%0A%0AI%20need%20a%20response%20by%3A%20%0A-%20Today%20-%20because%20the%20content%20needs%20to%20get%20published%20in%20the%20next%20deploy%0A-%20In%20the%20next%20few%20days%20-%20because%20I%20plan%20to%20publish%20this%20week%0A-%20Later%20on%20-%20I%E2%80%99m%20flexible%0A-%20This%20specific%20date%3A%20___%0A%0AHere%E2%80%99s%20some%20additional%20info%3A%20(Optional%2C%20but%20including%20these%20can%20help%20us%20understand%20and%20respond%20more%20quickly.)%0A-%20Attached%20screenshots%0A-%20Prod.cms.va.gov%20URL%0A-%20VA.gov%20URL%0A%0AThank%20you%2C%0A%0AYour%20name%0AYour%20title"
```
Comment (indytechcook) [inline docroot/modules/custom/va_gov_backend/js/script.js:13]: Holy hard-coded email batman...

**Comment by indytechcook** [comment]
It's great to update to a more standard JS!

File: docroot/modules/custom/va_gov_backend/js/script.js (hunk 3 above)
Comment (acrollet) [reply docroot/modules/custom/va_gov_backend/js/script.js:13]: I wonder if the JSD integration can obviate this... cc @oksana-c

**Review by ethanteague — DISMISSED** [review]
This is great, @acrollet - really appreciate you taking this on.

File: docroot/modules/custom/va_gov_backend/js/script.js (hunk 3 above)
Comment (acrollet) [reply docroot/modules/custom/va_gov_backend/js/script.js:13]: This is getting removed in #3399

File: docroot/modules/custom/va_gov_backend/js/script.js (hunk 1 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/js/script.js:5]: @acrollet @kevwalsh this looks to be related to an old implementation of alerts. There used to be a "read more" type of pattern in place that would reveal more text when clicked. This is no longer a thing.

Comment (acrollet) [reply docroot/modules/custom/va_gov_backend/js/script.js:5]: really appreciate you chasing that down @ethanteague !

**Comment by acrollet** [comment]
Noting that I need to document the build process before this can be merged.

**Comment by acrollet** [comment]
Docs updated, this is ready for review.

**Comment by acrollet** [comment]
I've re-validated the QA steps on the latest preview build.

**Comment by ndouglas** [comment]
I'm seeing some differences in GTM behavior between Lando and Prod, not sure if that's related or a side-effect of something else.  I'll have to compare with a different local branch, I think.

**Comment by ndouglas** [comment]
Seeing changes in the CSS but the `diff` tools I have are terrible enough that I can't actually determine what those changes are 🧐

**Comment by acrollet** [comment]
> Seeing changes in the CSS but the `diff` tools I have are terrible enough that I can't actually determine what those changes are 🧐

@ndouglas ah, I believe this is because the map files weren't generated in a previous PR. To fully test, run `gulp` and `npm run build` against master and this branch and check for differences.

**Comment by ndouglas** [comment]
Screen Shot 2020-12-17 at 10 24 59 AM

Yeah, that makes sense.

**Comment by acrollet** [comment]
@ndouglas I would work on improving the css build process, but I'd rather put the effort towards the new admin theme :-)

**Comment by ndouglas** [comment]
GTM is behaving consistently with `master`, as far as I can tell, so that checks the last checkbox I had personally.

**Comment by acrollet** [comment]
I re-verified the audience/topics JS (only changed js) after the latest merge of master.

**Comment by ethanteague** [comment]
> I re-verified the audience/topics JS (only changed js) after the latest merge of master.

Thanks for being all over this, @acrollet

**Comment by acrollet** [comment]
> Thanks for being all over this, @acrollet

@ethanteague

**Review by indytechcook — APPROVED** [review]
I tested good and bad commits.  The pre-commit hooks are fast and accurate.

Much better experience.

**Pull request merged**